nodewatcher/modules/analysis/channel_allocation/allocation_algorithms.py

```diff
@@ -0,0 +1,327 @@
+    :return: A dictionary with two k-v pairs, one for each frequency spectrum. Keys are "2.4GHz" and "5GHz".
+    """
+
+    highest_2ghz_channel = 11
+
+    nx_2ghz_graph = nx.Graph()
+    nx_5ghz_graph = nx.Graph()
+    channels_2ghz = {}
+    channels_5ghz = {}
+
+    for node in graph['v']:
+        if 'b' in node:
+            for bssid in node['b']:
+                if channel_lookup.is_2ghz_bssid(bssid):
+                    nx_2ghz_graph.add_node(bssid)
+                    # Relabel edges.
+                    for edge in graph['e']:
+                        if edge['f'] == node['i']:
+                            if edge['c'] <= highest_2ghz_channel:
+                                edge['f'] = bssid
+                else:
+                    nx_5ghz_graph.add_node(bssid)
+                    # Relabel edges.
+                    for edge in graph['e']:
+                        if edge['f'] == node['i']:
+                            if edge['c'] > highest_2ghz_channel:
+                                edge['f'] = bssid
+
+    for edge in graph['e']:
+        # TODO: read channel width from survey data.
+        edge['w'] = 20
+        if edge['c'] <= highest_2ghz_channel:
+            chosen_graph = nx_2ghz_graph
+            chosen_channels = channels_2ghz
+        else:
+            chosen_graph = nx_5ghz_graph
+            chosen_channels = channels_5ghz
+
+        chosen_graph.add_node(edge['t'])
+        chosen_graph.add_edge(
+            edge['f'],
+            edge['t'],
+            s=edge['s'],
+            c=channel_to_frequency(edge['c']),
+            n=edge['n'],
+            w=edge['w'],
+        )
+        if edge['t'] not in known_nodes:
+            chosen_channels[edge['t']] = {
+                'freq': channel_to_frequency(edge['c']),
+                'width': edge['w'],
+            }
+
+    # Run the algorithm on an appropriate graph.
+    optimal_2ghz_graph = greedy_color_with_constraints(nx_2ghz_graph, channels_2ghz)
+    optimal_5ghz_graph = greedy_color_with_constraints(nx_5ghz_graph, channels_5ghz)
+    channel_allocations = {}
+
+    # Combine results.
+    for node in optimal_2ghz_graph:
+        channel_allocations[node] = {
+            'freq': optimal_2ghz_graph[node]['freq'],
+            'width': optimal_2ghz_graph[node]['width'],
+            'interference': optimal_2ghz_graph[node]['interference'],
+            'current_interference': optimal_2ghz_graph[node]['current_interference'],
+        }
+
+    for node in optimal_5ghz_graph:
+        channel_allocations[node] = {
+            'freq': optimal_5ghz_graph[node]['freq'],
+            'width': optimal_5ghz_graph[node]['width'],
+            'interference': optimal_5ghz_graph[node]['interference'],
+            'current_interference': optimal_5ghz_graph[node]['current_interference'],
+
+        }
+
+    return channel_allocations
+
+
+def channel_to_frequency(channel):
+    ch_to_freq = {
+        '1': 2412,
```

Comment:
Why not using that algorithm I linked to you?

Also, this function expects inputs as strings? It would be great if you would make sure it works with numbers as well. It is a bid strange API in this way.